AdyenCard/Components/Card/CardViewController.swift

```diff
@@ -409,7 +407,33 @@ extension CardViewController: CardViewControllerProtocol {
 
 extension CardViewController {
     private func openCardScanner() {
-        // TODO: Use CardScanning module
+        let scannerNavigationController = UINavigationController()
+        if let scannerViewController = AdyenCardScanner.CardScanner.createCardScanner(completion: { [weak self] result in
+            switch result {
+            case let .success(card):
+                scannerNavigationController.dismiss(animated: true)
+                self?.processScannedCard(card)
+            case let .failure(error):
+                // TODO: Error handling
+                print("Scanner error: \(error)")
+            }
+        }) {
+            scannerNavigationController.setViewControllers(
+                [scannerViewController],
+                animated: false
+            )
+            presentViewController(scannerNavigationController, animated: true)
+        }
+    }
+    
+    private func processScannedCard(_ card: AdyenCardScanner.CreditCard) {
+        items.numberContainerItem.numberItem.value = card.number ?? ""
+        
+        if let expirationDate = card.expirationDate {
+            items.expiryDateItem.setExpiryDate(expirationDate)
```

Comment:
Any reason not to use the nil coalescing operator here? (Like you did with the numberItem.value)

Reply:
Passing an optional date into `setExpiryDate()`? Let me know if I follow this right

Reply:
Currently `AdyenCardScaning` can return `nil` for expiry date, this is why this check belongs here in `CardViewController` as it captures the rule "We do not nullify expiry date if i was entered before as we do not always expect to have it".

@nauaros Can we make card number non-nil in the `AdyenCardScanner` API so it captures framework behavior via API?

Reply:
Great comment Alex, I removed `else` part :)